recipes-extended/baseboxd-tools/baseboxd-tools/bundle-debug-info

```diff
@@ -190,6 +191,42 @@ function get_port_list() {
   log_cmd_output onlpdump -S
 }
 
+function get_mstpd_state() {
+  title "Retrieving MSTPD state"
+  OUTPUT_FILE="$TMPDIR/mstpd_state"
+  LST="Service State" log_cmd_output /bin/systemctl status mstpd
```

Comment:
log_cmd_output aborts bundle-debug-info (intentionally) if a log command fails. "systemctl status" returns a non-zero exit code if the service is not running.

If you don't want to bypass log_cmd_output entirely, you can do something like:

```
if systemctl is-active --quiet mstpd; then
  LST="Service State" log_cmd_output /bin/systemctl status mstpd
fi
```

Maybe throw in `journalctl -u mstpd` if you want to capture the log for an inactive mstpd service.

Reply:
> log_cmd_output aborts bundle-debug-info (intentionally) if a log command fails. "systemctl status" returns a non-zero exit code if the service is not running.

Huh, I would have expected to just abort the current part. Is there any way to avoid this?

> Maybe throw in journalctl -u mstpd if you want to capture the log for an inactive mstpd service.

We already capture the full journal by default, so I don't think we need to create a separate log for mstpd (we don't either for baseboxd).

Reply:
> Huh, I would have expected to just abort the current part. Is there any way to avoid this?

I am not sure what you mean by "this". Failing immediately and hard catches errors in testing that we might overlook otherwise. I already posted code to avoid an error in this specific case. You can also not use the log_cmd_output function and write the output directly. We could also expand log_cmd_output (or report_result) with an option to ignore the exit status of commands, but I'm not convinced that we should.

Reply:
With "this" I mean I would like to capture the output of systemctl status mstpd, regardless of the exit code of it.

Reply:
Thinking about it, I think it would be better if we didn't abort on non-zero return values. While it's nice for verification that we do things correctly, I think for customers this is the opposite of nice. They will want to collect debug info especially if things don't work properly, but if the collect-debug-info script aborts when things don't work properly, then they can't give us any.

Reply:
Made it possible to ignore errors (both globally and per-command).